tests/dags/common/test_slack.py

```diff
@@ -336,3 +341,19 @@ def test_on_failure_callback_does_nothing_without_hook(http_hook_mock):
     http_hook_mock.get_conn.side_effect = AirflowNotFoundException("nope")
     on_failure_callback({})
     http_hook_mock.run.assert_not_called()
+
+
+@pytest.mark.parametrize(
+    "environment, force_slack_alert, expected_result",
+    [
+        ("dev", False, False),
+        ("dev", True, True),
+        ("prod", False, True),
+        ("prod", True, True),
+    ],
+)
+def test_should_send_message(environment, force_slack_alert, expected_result):
+    with mock.patch("common.slack.Variable") as MockVariable:
+        # Mock the calls to Variable.get, in order
+        MockVariable.get.side_effect = [environment, force_slack_alert]
```

Comment:
This is perfectly fine as-is, just noting for pedagogical reasons:
The other way I've seen something simple like this done is to create a dictionary and then use the `__getitem__` call as the actual function:
```suggestion
        d = {"environment": environment, "force_slack_alert": force_slack_alert}
        MockVariable.get.side_effect = d.__getitem__
```

This makes the test robust to changes in the call order to `Variable`.